joblib/numpy_pickle.py

```diff
@@ -70,13 +73,20 @@ class NumpyArrayWrapper(object):
         Default: False.
     """
 
-    def __init__(self, subclass, shape, order, dtype, allow_mmap=False):
+    def __init__(self, subclass, shape, order, dtype, allow_mmap=False,
+                 array_bytes_aligned=True):
         """Constructor. Store the useful information for later."""
         self.subclass = subclass
         self.shape = shape
         self.order = order
         self.dtype = dtype
         self.allow_mmap = allow_mmap
+        self.array_bytes_aligned = array_bytes_aligned
+
+    def _array_bytes_aligned(self):
+        # joblib <= 1.1 pickles NumpyArrayWrapper instances don't have an
+        # array_bytes_aligned attribute
+        return getattr(self, 'array_bytes_aligned', False)
```

Comment:
I think if we want to change our mind about `NUMPY_ARRAY_ALIGNEMENT_BYTES` we need to have an attribute of NumpyArrayWrapper so that old pickles (i.e. with the previous alignment e.g. 16) can be unpickled correctly (i.e. without using the new alignment e.g. 64)

Reply:
OK I have added `numpy_array_alignment_bytes` as a `NumpyArrayWrapper` to allow us to change our mind about the default alignment in joblib pickles.